modular_zubbers/code/modules/mouseover/highlights.dm

```diff
@@ -0,0 +1,143 @@
+	// and images have vis contents - so we throw a null image into client.images, then
+	// throw a holder object with the appearance of the mouse-overed atom into its vis contents.
+	mouseover_highlight_dummy.appearance = AM
+	mouseover_highlight_dummy.name = ""
+	mouseover_highlight_dummy.verbs.Cut()
+	mouseover_highlight_dummy.vis_flags |= VIS_INHERIT_ID
+	mouseover_highlight_dummy.dir = AM.dir
+	mouseover_highlight_dummy.transform = AM.transform
+
+	// For some reason you need to explicitly zero the pixel offsets of the holder object
+	// or anything with a pixel offset will not line up with the highlight. Thanks DM.
+	mouseover_highlight_dummy.pixel_x = 0
+	mouseover_highlight_dummy.pixel_y = 0
+	mouseover_highlight_dummy.pixel_w = 0
+	mouseover_highlight_dummy.pixel_z = 0
+
+	// Replane to be over the UI, make sure it can't block clicks, and set its outline.
+	mouseover_highlight_dummy.mouse_opacity = 0
+	mouseover_highlight_dummy.layer = ABOVE_HUD_PLANE
+	mouseover_highlight_dummy.plane = BALLOON_CHAT_PLANE
+	mouseover_highlight_dummy.alpha = 255
+	mouseover_highlight_dummy.appearance_flags |= (KEEP_TOGETHER|RESET_COLOR)
+	mouseover_highlight_dummy.add_filter(
+		"drop",
+		1,
+		list(
+			type = "drop_shadow",
+			color = mouseover_rainbow ? rgb(rand(0,255),rand(0,255),rand(0,255)) : "#0082c6",
+			size = 1,
+			offset = 2, x = 0, y = 0
+		)
+	)
+	animate(mouseover_highlight_dummy, pixel_y = 24, time = 0.5 SECONDS, easing = ELASTIC_EASING, alpha = 180, maptext_y = 24)
+
+	mouseover_highlight_dummy.maptext_width = 128
+	mouseover_highlight_dummy.maptext_x = -48
+	mouseover_highlight_dummy.maptext = MAPTEXT_SPESSFONT(AM)
+	// Finally update our highlight's vis contents and location .
+	clear_vis_contents(current_highlight)
+	add_vis_contents(current_highlight, mouseover_highlight_dummy)
+	current_highlight.loc = object
+	current_highlight_atom = WEAKREF(AM)
+
+	// Keep track our params so the update ticker knows if we were holding shift or not.
+	last_mouseover_params = params
+
+	return TRUE
+
+// Simple hooks to catch the client mouseover/mouseleave events and start our highlight timer as needed.
+/client/MouseEntered(object, location, control, params)
+	if(world.time > last_mouseover_highlight_time && refresh_mouseover_highlight(object, params, check_adjacency = TRUE) && !mouseover_refresh_timer)
+		last_mouseover_highlight_time = world.time
+		mouseover_refresh_timer = addtimer(mouseover_callback, 1 SECONDS, (TIMER_OVERRIDE | TIMER_UNIQUE | TIMER_STOPPABLE))
+	. = ..()
+/client/MouseExited(object, location, control, params)
+	var/initalpha = initial(mouseover_highlight_dummy.alpha)
+	var/atom/movable/current_atom = current_highlight_atom?.resolve()
+	if(current_atom != object && mouseover_highlight_dummy)
+		animate(mouseover_highlight_dummy, pixel_y = 0, time = 0.2 SECONDS, easing = ELASTIC_EASING, alpha = initalpha)
+		refresh_mouseover_highlight_timer(current_atom, object)
+
+	. = ..()
+
+
```

Comment:
`/datum/client_interface/var/mouseover` is missing from this file, which is causing runtimes!